SOURCES/etc/xapi.d/plugins/raid.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from functools import wraps
+import sys
+import json
+
+import XenAPIPlugin
+
+sys.path.append('.')
+from xcpngutils import configure_logging, run_command
+from xcpngutils.filelocker import FileLocker
+
+_LOGGER = configure_logging('raid')
+
+
+def lock(*pid_args, **pid_kwargs):
+    def wrapper(func):
+        @wraps(func)
+        def decorator(*args, **kwargs):
+            try:
+                with FileLocker(*pid_args, **pid_kwargs):
+                    return func(*args, **kwargs)
+            except Exception as e:
+                return json.dumps({'error': str(e)})
```

Comment:
This is not a fatal error, rather a foreseeable failure due to an operation in progress. In updater.py we simply raise an exception with a human-readable error text so that XO logs are meaningful to users.

Reply:
I'm struggling a bit to wrap the lock exception without wrapping the whole `with` block.